core/src/ast/enums.rs

```diff
@@ -35,11 +36,10 @@ impl From<&syn::ItemEnum> for Enum {
                         .discriminant
                         .as_ref()
                         .map(|d| {
-                            if let syn::Expr::Lit(syn::ExprLit {
-                                attrs: _,
-                                lit: syn::Lit::Int(lit_int),
-                            }) = &d.1
-                            {
+                            // Reparsing, signed literals are represented
+                            // as a negation expression
+                            let lit: Result<syn::Lit, _> = syn::parse2(d.1.to_token_stream());
```

Comment:
Thought: I think it would be a bit cleaner to instead match on the unary operation without going back and forth to a token stream

Reply:
I started with that but the code looks much gnarlier, and jumping between token streams isn't super expensive